code/_onclick/hud/parallax.dm

```diff
@@ -176,81 +181,80 @@ var/list/parallax_on_clients = list()
 		pixel_y += 32 * round(i/15)
 
 	for(var/i in 0 to 8)
-		var/obj/screen/parallax/parallax_layer = new /obj/screen/parallax()
+		var/obj/screen/parallax/parallax_layer = getFromPool(/obj/screen/parallax)
 
 		var/list/L = list()
 		for(var/j in 1 to 225)
 			if(plane1[j+i*225] <= PARALLAX4_ICON_NUMBER)
 				var/image/I = image('icons/turf/space_parallax4.dmi',"[plane1[j+i*225]]")
 				I.pixel_x = pixel_x[j]
 				I.pixel_y = pixel_y[j]
-				I.plane = PLANE_SPACE_PARALLAX
 				L += I
 
 		parallax_layer.overlays = L
 		parallax_layer.parallax_speed = 0
 		parallax_layer.plane = PLANE_SPACE_PARALLAX
-		calibrate_parallax(parallax_layer,i+1)
+		parallax_layer.calibrate_parallax(i+1)
 		parallax_icon[index] = parallax_layer
 		index++
 
 	for(var/i in 0 to 8)
-		var/obj/screen/parallax/parallax_layer = new /obj/screen/parallax()
+		var/obj/screen/parallax/parallax_layer = getFromPool(/obj/screen/parallax)
 
 		var/list/L = list()
 		for(var/j in 1 to 225)
 			if(plane2[j+i*225] <= PARALLAX3_ICON_NUMBER)
 				var/image/I = image('icons/turf/space_parallax3.dmi',"[plane2[j+i*225]]")
 				I.pixel_x = pixel_x[j]
 				I.pixel_y = pixel_y[j]
-				I.plane = PLANE_SPACE_PARALLAX
 				L += I
 
 		parallax_layer.overlays = L
 		parallax_layer.parallax_speed = 1
 		parallax_layer.plane = PLANE_SPACE_PARALLAX
-		calibrate_parallax(parallax_layer,i+1)
+		parallax_layer.calibrate_parallax(i+1)
 		parallax_icon[index] = parallax_layer
 		index++
 
 	for(var/i in 0 to 8)
-		var/obj/screen/parallax/parallax_layer = new /obj/screen/parallax()
+		var/obj/screen/parallax/parallax_layer = getFromPool(/obj/screen/parallax)
 		var/list/L = list()
 		for(var/j in 1 to 225)
 			if(plane3[j+i*225] <= PARALLAX2_ICON_NUMBER)
 				var/image/I = image('icons/turf/space_parallax2.dmi',"[plane3[j+i*225]]")
 				I.pixel_x = pixel_x[j]
 				I.pixel_y = pixel_y[j]
-				I.plane = PLANE_SPACE_PARALLAX
 				L += I
 
 		parallax_layer.overlays = L
 		parallax_layer.parallax_speed = 2
 		parallax_layer.plane = PLANE_SPACE_PARALLAX
-		calibrate_parallax(parallax_layer,i+1)
+		parallax_layer.calibrate_parallax(i+1)
 		parallax_icon[index] = parallax_layer
 		index++
 
 	parallax_initialized = 1
 
-/proc/calibrate_parallax(var/obj/screen/parallax/p_layer,var/i)
-	if(!p_layer || !i) return
+/obj/screen/parallax/proc/calibrate_parallax(var/i)
+	if(!i) return
 
 	/* Placement of screen objects
 	1	2	3
 	4	5	6
 	7	8	9
 	*/
+	base_offset_x = -7*32
```

Comment:
You could replace 
`\s*(([A-Za-z0-9_-]+\.)*)view\s*=\s*(.+)`
With 
`$1set_view($3)`

There's no dot because the capture group includes it and a plan view= doesn't need one.
Won't handle increment or decrement, but that's something else.